### ISSUE TITLE
PRDT: Build out API

### ISSUE THREAD

**Issue by Dianadec** [bcgov/reserve-rec-api#1]
#### Description:
- building out the API is minimal work now due to the other work that has been done
- so this ticket is about a search endpoint that ties into the opensearch cluster
- This iteration is just about pulling data from the backend
- Second iteration will be more complex

#### Acceptance Criteria:

- [x] Create an endpoint `api/search` that queries the database and returns results
- [x] NOTE: - build this ticket more after the data modelling ticket is sorted

#### Development Checklist:

- [ ] Look into endpoint documentation (swagger)   -> this is now in a new ticket: https://github.com/bcgov/reserve-rec-api/issues/16 (not super critical for operation of the system, but it is a documentation piece).
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by #3 
- Blocking
- Related to #9 

**Relevant documentation as reference**

**Definition of Ready**

- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [ ] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by Dianadec** [comment]
Refinement notes:
- need list of endpoints
- build this ticket more after the data model

**Comment by Dianadec** [comment]
Refinement notes:
- building out the API is minimal work now due to the other work that has been done
- so this ticket is about a search endpoint that ties into the opensearch cluster
- This iteration is just about pulling data from the backend
- Second iteration will be more complex

**Comment by Dianadec** [comment]
Todo - create new ticket to cover the following:

- Configure keycloak clients into keycloak test and prod
- go into opensearch dashboard to wire up some of those controls to connect to the keycloak clients from previous step in test and prod
-
